nixos/modules/services/mail/maddy.nix

```diff
@@ -220,20 +233,14 @@ in {
       '';
     };
 
-    users.users = optionalAttrs (cfg.user == "maddy") {
-      maddy = {
-        description = "Maddy service user";
-        group = cfg.group;
-        home = "/var/lib/maddy";
-        createHome = true;
-        isSystemUser = true;
-      };
+    users.users.${cfg.user} = {
+      description = "Maddy mail transfer agent user";
+      uid = config.ids.uids.maddy;
+      group = cfg.group;
```

Comment:
This isn't a great pattern - and you don't see it very often in NixOS. You can easily run into conflicts if you apply settings to arbitrary users, and you often override what users want/expect. It is best to conditionally provision users and groups based on whether the default value has been selected or not (ie. `users.users = optionalAttrs (cfg.user == "maddy") {`).

Instead of this change, I would suggest you improve the description of the `user` and `group` options, as per https://github.com/NixOS/nixpkgs/pull/148043.

Reply:
Oh this is a very good recommendation! Thanks